rmf_fleet_adapter/include/rmf_fleet_adapter/agv/RobotUpdateHandle.hpp

```diff
@@ -110,6 +111,54 @@ class RobotUpdateHandle
   /// value that was given to the setter.
   rmf_utils::optional<rmf_traffic::Duration> maximum_delay() const;
 
+  /// The ActionExecution class should be used to manage the execution of and
+  /// provide updates on ongoing actions.
+  class ActionExecution
+  {
+  public:
+    // Update the amount of time remaining for this action
+    void update_remaining_time(rmf_traffic::Duration remaining_time_estimate);
+
+    // Trigger this when the action is finished
+    void finished();
+
+    // Returns false if the Action has been killed or cancelled
+    bool okay() const;
+
+    // TODO: Consider giving access to the participant schedule and
+    // traffic negotiation
+
+    ~ActionExecution()
```

Comment:
We should put the definition of this function into the proper translation unit, i.e. `RobotUpdateHandle.cpp`. I understand you were probably copying the example code that I provided, but I defined it inline for brevity. It's not actually good to leave this function definition exposed in a public API, because that can lead to poorly defined behavior if we ever change the implementation of the function.